src/commands/git.rs

```diff
@@ -452,10 +455,18 @@ fn cmd_git_clone(
 fn do_git_clone(
     ui: &mut Ui,
     command: &CommandHelper,
+    colocate: bool,
     source: &str,
     wc_path: &Path,
 ) -> Result<(WorkspaceCommandHelper, Option<String>), CommandError> {
-    let (workspace, repo) = Workspace::init_internal_git(command.settings(), wc_path)?;
+    let (workspace, repo) = if colocate {
+        let git_path = &wc_path.join(".git");
+        // Need to clone the git repo first.
+        git2::Repository::init_bare(git_path)?;
```

Comment:
Bare repos don't have a working copy (called a working tree in git), so will this really created a colocated workspace? I think we need some tests showing that the doing a `git checkout` (via libgit2) updates the files in the working copy.

Reply:
You're correct.

Reply:
I changed it to use `git2::Repository::init`.  I've test manually but I'm not quite sure how to write the test you suggest.

Reply:
Actually, we can instead test that the git repo is checked out to the right commit. You can use `git_repo.head()` to find what HEAD points to (it should point to the same commit as the jj repo). That's probably enough to check.

Reply:
I'm not sure you noticed my suggestion here. I unresolved it to make it more visible.

Reply:
I feel like I'm causing you more work than I'm solving with this PR.

I tried your suggestion, but it fails even for the regular clone case:

```

    let jj_git_repo = git2::Repository::open(
        test_env.env_root().join("clone")).expect("Could not open clone repo");
    assert_eq!(
        jj_git_repo.head().expect("Clone Repo HEAD should be set.").name(), 
        git_repo.head().expect("Repo HEAD should be set.").name());
```

`jj_git_repo.head()....name() == "/refs/heads/master")`

Reply:
Done, let me know if this is not what you had in mind and thanks for helping me ramp up on jj code.